src/test/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicatorTest.java

```diff
@@ -0,0 +1,231 @@
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandSlowShutterOff() {
+		String status = (String) averPTZCommunicator.digestResponse(ReplyPacket.AUTO_SLOW_SHUTTER_OFF.getCode(), 1, CommandType.INQUIRY, Command.AUTO_SLOW_SHUTTER.getName());
+		Assert.assertEquals("Off", status);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect shutter speed value is 15
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandShutterSpeed() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.SHUTTER_SPEED.getCode(), 1, CommandType.INQUIRY, Command.SHUTTER_DIRECT.getName());
+		Assert.assertEquals(15, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect iris level value is 13
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandIrisLevel() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.IRIS_LEVEL.getCode(), 1, CommandType.INQUIRY, Command.IRIS_DIRECT.getName());
+		Assert.assertEquals(13, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect gain level value is 45
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandGainLevel() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.GAIN_LEVEL.getCode(), 1, CommandType.INQUIRY, Command.GAIN_DIRECT.getName());
+		Assert.assertEquals(45, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect RGain value is 33
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandRGainValue() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.RGAIN.getCode(), 1, CommandType.INQUIRY, Command.RGAIN.getName());
+		Assert.assertEquals(33, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect BGain value is 52
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandBGainValue() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.BGAIN.getCode(), 1, CommandType.INQUIRY, Command.BGAIN.getName());
+		Assert.assertEquals(52, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect gain limit level value is 8
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandGainLimitLevel() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.GAIN_LIMIT_LEVEL.getCode(), 1, CommandType.INQUIRY, Command.GAIN_LIMIT_DIRECT.getName());
+		Assert.assertEquals(8, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect exposure value is 9
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandExposureValue() {
+		int value = (int) averPTZCommunicator.digestResponse(ReplyPacket.EXPOSURE_VALUE.getCode(), 1, CommandType.INQUIRY, Command.EXP_COMP_DIRECT.getName());
+		Assert.assertEquals(9, value);
+	}
+
+	/**
+	 * Test AverPTZCommunicator#digestResponse success
+	 * Expect last preset recalled value is 2
+	 */
+	@Test
+	public void testDigestResponseInquiryCommandLastPresetRecalled() {
```

Comment:
This test is not correct, I think. We will get the null pointer exception because we do not support PRESET in digestResponse method.

Reply:
as I said above :D I forgot add preset in digestResponse method